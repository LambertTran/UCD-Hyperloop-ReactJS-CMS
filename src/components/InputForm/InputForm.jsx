/** Packages */
import React from 'react';
import { Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';

/** Actions */
import { uploadImage } from '../../actions/postRequest';

export class InputForm extends React.Component {

  renderField = (field) => {
    // delete the value field
    delete field.input.value

    return (
      <div className='form-group has-danger'>
        <label>{field.label} </label>
        <input 
          className='form-control'
          type={field.type}
          {...field.input}
        />
        <div className='text-help'>
          { (field.meta.error && field.meta.touched) ? field.meta.error : '' }
        </div>
      </div> 
    );
  }

  onSubmit = (data) => {
    this.props.uploadImage(data.img[0])
  }

  render() {

    const { handleSubmit } = this.props;

    return (
      <div className='upload-modal-content'>
        <form onSubmit = {handleSubmit(this.onSubmit.bind(this))}>
          <h3>Please choose 1 image for your team</h3>
          <Field label='Image' name='img' component={this.renderField} type='file'/>
          <button type='submit' className='btn btn-primary'>Upload</button>
          <button type='button' className='btn btn-danger'>Cancel</button>
        </form>
      </div>
    )
  }
}

// validate the values object got from redux form
// after user submit the form
function validate(data) {
  const errors = {};

  if (!data.img || !data.img[0]) {
    errors.img  = 'Missing image file'; 
  } else if (data.img[0]) {
    let fileType = data.img[0].name.split('.');
    switch (fileType.pop()) {
      case 'png':
        break;
      case 'jpg':
        break;
      case 'jpeg':
        break;
      default:
        errors.img = 'Must be an image file'

    }
  }
  return errors;
} 

export default reduxForm({
  validate,
  form: 'PostsNewForm',
})(
  connect(null,{ uploadImage })(InputForm)
)