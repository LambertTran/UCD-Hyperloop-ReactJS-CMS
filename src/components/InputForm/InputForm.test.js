// packages
import React from 'react';
import { shallow } from 'enzyme';
import setupTests from '../../shares/setupTests';

import InputForm from './InputForm';

describe('InputForm', () => { 
  const inputForm = shallow( <InputForm />);

  it('renders correctly', () => {
    expect(inputForm).toMatchSnapshot();
  })

  it('has default opacity is 0', () => {
    expect(inputForm.state().opacity).toEqual(0);
  })

  it('should close when cancel button is clicked', () => {
    inputForm.find('.btn-danger').simulate('click');
    expect(inputForm.state().opacity).toEqual(0);
  })

  it('should open when cancel button is clicked', () => {
    inputForm.find('.btn-primary').simulate('click');
    expect(inputForm.state().opacity).toEqual(1);
  })
})