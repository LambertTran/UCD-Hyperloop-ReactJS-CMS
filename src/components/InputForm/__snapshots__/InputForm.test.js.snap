// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action="/admin/team/{{team}}/upload-image"
        encType="multipart/form-data"
        method="post"
>
        <h3>
                Please choose 1 image for your team
        </h3>
        <div
                className="form-group"
        >
                <label>
                        updates
                </label>
                <input
                        className="form-control"
                        name="img"
                        type="file"
                />
        </div>
        <button
                className="btn btn-primary"
                onClick={undefined}
                type="submit"
        >
                Upload
        </button>
        <button
                className="btn btn-danger"
                onClick={[Function]}
                type="button"
        >
                Cancel
        </button>
</form>,
      "className": "upload-modal-content",
      "style": Object {
        "opacity": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "/admin/team/{{team}}/upload-image",
        "children": Array [
          <h3>
            Please choose 1 image for your team
</h3>,
          <div
            className="form-group"
>
            <label>
                        updates
            </label>
            <input
                        className="form-control"
                        name="img"
                        type="file"
            />
</div>,
          <button
            className="btn btn-primary"
            onClick={undefined}
            type="submit"
>
            Upload
</button>,
          <button
            className="btn btn-danger"
            onClick={[Function]}
            type="button"
>
            Cancel
</button>,
        ],
        "encType": "multipart/form-data",
        "method": "post",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please choose 1 image for your team",
          },
          "ref": null,
          "rendered": "Please choose 1 image for your team",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                updates
</label>,
              <input
                className="form-control"
                name="img"
                type="file"
/>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "updates",
              },
              "ref": null,
              "rendered": "updates",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "name": "img",
                "type": "file",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Upload",
            "className": "btn btn-primary",
            "onClick": undefined,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Upload",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "className": "btn btn-danger",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action="/admin/team/{{team}}/upload-image"
          encType="multipart/form-data"
          method="post"
>
          <h3>
                    Please choose 1 image for your team
          </h3>
          <div
                    className="form-group"
          >
                    <label>
                              updates
                    </label>
                    <input
                              className="form-control"
                              name="img"
                              type="file"
                    />
          </div>
          <button
                    className="btn btn-primary"
                    onClick={undefined}
                    type="submit"
          >
                    Upload
          </button>
          <button
                    className="btn btn-danger"
                    onClick={[Function]}
                    type="button"
          >
                    Cancel
          </button>
</form>,
        "className": "upload-modal-content",
        "style": Object {
          "opacity": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "/admin/team/{{team}}/upload-image",
          "children": Array [
            <h3>
              Please choose 1 image for your team
</h3>,
            <div
              className="form-group"
>
              <label>
                            updates
              </label>
              <input
                            className="form-control"
                            name="img"
                            type="file"
              />
</div>,
            <button
              className="btn btn-primary"
              onClick={undefined}
              type="submit"
>
              Upload
</button>,
            <button
              className="btn btn-danger"
              onClick={[Function]}
              type="button"
>
              Cancel
</button>,
          ],
          "encType": "multipart/form-data",
          "method": "post",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please choose 1 image for your team",
            },
            "ref": null,
            "rendered": "Please choose 1 image for your team",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  updates
</label>,
                <input
                  className="form-control"
                  name="img"
                  type="file"
/>,
              ],
              "className": "form-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "updates",
                },
                "ref": null,
                "rendered": "updates",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control",
                  "name": "img",
                  "type": "file",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Upload",
              "className": "btn btn-primary",
              "onClick": undefined,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Upload",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cancel",
              "className": "btn btn-danger",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Cancel",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
